python/lsst/daf/butler/core/datastore.py

```diff
@@ -63,19 +65,17 @@ class DatastoreTransaction:
     ----------
     parent : `DatastoreTransaction`, optional
         The parent transaction (if any)
-
-    Attributes
-    ----------
-    parent : `DatastoreTransaction`
-        The parent transaction.
     """
-    Event = Event
+    Event: ClassVar[Type] = Event
+
+    parent: Optional['DatastoreTransaction']
+    """The parent transaction. (`DatastoreTransaction`, optional)"""
 
     def __init__(self, parent=None):
         self.parent = parent
         self._log = []
 
-    def registerUndo(self, name, undoFunc, *args, **kwargs):
+    def registerUndo(self, name: str, undoFunc: Callable, *args: Any, **kwargs: Any) -> None:
```

Comment:
I have been leaving off the return type annotation for functions that always return `None`, but I don't have a strong opinion on that.  Should we establish a convention one way or the other?